pose_calculator.py

```diff
@@ -125,17 +119,119 @@ def update_values(self, r, t):
         self.previous_r.append(r)
         self.previous_t.append(t)
 
+    def get_contour_center(self, contour):
+        x, y, w, h = cv2.boundingRect(contour)
+        cx = x + w / 2
+        cy = y + h / 2
+        return cx, cy
+
+    # Subtracts floor from current depth frame, returns new depth frame with removed floor values
+    def subtract_floor(self, depth):
+        floor = self.floor_frame.copy()
+        floor[floor > 4] = 0
+        subtracted = np.where(abs(depth - floor) < 0.05, 0, depth)
+        return subtracted
+
+    # Finds obstacles, returns list of obstacle contours with area greater than 1000 pixels
+    def find_obstacles(self, depth, max_distance):
+        if depth is None:
+            return None
+
+        removed_floor = self.subtract_floor(depth)
+        cv2.imshow("subtracted", removed_floor)
+
+        low_obstacle_distance = 0.56
+        high_obstacle_distance = float(max_distance) - 0.05
+        mask = cv2.inRange(removed_floor, low_obstacle_distance, high_obstacle_distance)
```

Comment:
We made an obstacle any contour that has an area above 1000 and is between `low_obstacle_distance` and `high_obstacle_distance` meters away, with `high_obstacle_distance` being the distance of the closest ball (if there is one; otherwise it's 3 meters). This should give any object that is blocking the path to balls or blocking the way of the robot.